### ISSUE TITLE
Повторная отправка формы

### ISSUE THREAD

**Issue by alex-eri** [nightflyza/Ubilling#71]
При неистовом нажатии некоторых кнопок форма в админке может примениться несколько раз до перезагрузки страницы.
Например, в модуле addcash это приводит к зачислению двойной суммы на счет пользователю.

Посмотрел сейчас в коде - не нашел защиты от этого.  Нужно что-то вроде xsrf токена вешать.

**Comment by alex-eri** [comment]
С другой стороны можно дисаблить форму яваскриптом до завершения аякса или на совсем в случае релоада странички.

**Comment by nightflyza** [comment]
Да, мы в курсе.  К сожалению небыло пока времени заниматься нормальной поддержкой "неистовости". Все что происходит на эту тему на данный момент, это принудительный редирект на себя, с целью очистки POST данных форм и предотвращения их повторной отправки.
Если у вас есть реальное желание помочь - велкам, пуллреквесты как всегда приветствуются.

**Comment by nightflyza** [comment]
> С другой стороны можно дисаблить форму яваскриптом до завершения аякса или на совсем в случае релоада странички.

Есть идеи как это сделать с сохранением легаси, в случае, если мы не знаем elementid ниодной из форм?

**Comment by alex-eri** [comment]
"не знаем elementid" - для форм с экшаном можно закостылить яваскрипт, который будет ждать появление форм в DOMе и вешать на них обработчик на onsubmit. С аяксом так не выйдет. там прийдется лезть в каждый обработчик аякса. Тут много на аяксе?

Редирект на себя - отличная идея. Это не поможет от неистовости, а только от обновления странички.

**Comment by alex-eri** [comment]
Попробую закостылить яваскипт.

**Comment by nightflyza** [comment]
> "не знаем elementid" - для форм с экшаном можно закостылить яваскрипт, который будет ждать появление форм в DOMе и вешать на них обработчик на onsubmit. 

Звучит кошерно

> Тут много на аяксе?
Вообще в полутора местах, и там такие вещи не особо замечены.

> Редирект на себя - отличная идея. Это не поможет от неистовости, а только от обновления странички.
Ну вот на данный момет это и есть вся наша поддержка "неистовости".

> Попробую закостылить яваскипт.
Было бы просто отлично

**Comment by alex-eri** [comment]
не мержи, там косяк)

**Comment by alex-eri** [comment]
Косяк исправил, формочки блокирует ;)

**Comment by nightflyza** [comment]
UPD: вынес загрузку скрипта в wf_FormDisabler().  Формочки так же дисабляться при конкатенации результата к форме. Пока что добавил в addcash и userreg.
